hspec-core/src/Test/Hspec/Core/Spec/Monad.hs

```diff
@@ -32,16 +32,36 @@ import           Test.Hspec.Core.Tree
 
 import           Test.Hspec.Core.Config.Definition (Config)
 
+-- | A `SpecWith` that can be evaluated directly by the
+-- `Test.Hspec.Core.Runner.hspec` function as it does not require any
+-- parameters.
 type Spec = SpecWith ()
 
+-- | `SpecM` with an empty `ReaderT` environment.
+--
+-- This is the most commonly used spec tree writer type.
+-- It is parameterized over the type @a@, which is the argument type provided
+-- to function-like `Example` instances.
+--
+-- To supply an argument to `SpecWith` tests to turn them into `Spec`, use
+-- `Test.Hspec.Core.Hooks.around` or `Test.Hspec.Core.Hooks.mapSubject`.
+--
+-- Values of this type are created by `Test.Hspec.Core.Spec.it`,
+-- `Test.Hspec.Core.Spec.describe` and similar.
 type SpecWith a = SpecM a ()
 
 -- |
 -- @since 2.10.0
 modifyConfig :: (Config -> Config) -> SpecWith a
 modifyConfig f = SpecM $ tell (Endo f, mempty)
 
--- | A writer monad for `SpecTree` forests
+-- | A writer monad for `SpecTree` forests.
+--
+-- This is the monad type passed to `Test.Hspec.Core.Spec.describe` and is used
+-- to construct the forest of trees of spec items.
```

Comment:
```suggestion
-- to construct the forest of spec items.
```

A forest always consists of trees. Does it do anything from the perspective of a user to mention this explicitly here?